Sources/MongoMobile/MongoMobile.swift

```diff
@@ -5,31 +5,53 @@ import mongo_embedded
 public struct MongoClientSettings {
     /// the database path to use
     public let dbPath: String
-
+    
     /// public member-wise initializer
     public init(dbPath: String) {
         self.dbPath = dbPath
     }
 }
 
-/// Errors associated with `MongoMobile`.
 public enum MongoMobileError: LocalizedError {
     case invalidClient()
     case invalidInstance(message: String)
     case invalidLibrary()
     case instanceDropError(message: String)
     case cleanupError(message: String)
+}
+
+/// Replica enum of mongo_embedded_v1_error typedef
```

Comment:
I'm confused, were these not synthesized by the binding? I thought @kmahar was using them in her first patch

Reply:
I'm not sure what you mean?

Reply:
so rather than having to duplicate the enum in Swift, we should be able to just use the original one from `mongo_embedded`.

looking at libmongocapi in Xcode, the generated Swift interface for `mongo_embedded_v1_error` is:

```swift
public struct mongo_embedded_v1_error : Equatable, RawRepresentable {

    public init(_ rawValue: Int32)

    public init(rawValue: Int32)

    public var rawValue: Int32
}
```

so basically you can go from Swift value -> enum by doing something like `mongo_embedded_v1_error(rawValue: 0)`, and do the reverse with something like `MONGO_EMBEDDED_V1_ERROR_EXCEPTION.rawValue` as I was doing [here](https://github.com/mongodb/swift-mongo-mobile/blob/6ca646f4752ed504bebbe995c4963ae142c62b37/Sources/MongoMobile/MongoMobile.swift#L88).

and the Swift interface for one of the places where we get a `result` back:
```swift
public func mongo_embedded_v1_instance_destroy(_ instance: OpaquePointer!, _ status: OpaquePointer!) -> Int32
```

so I *think* it should be possible to just directly store a `mongo_embedded_v1_error` in `MongoEmbeddedV1Error`, and you can use the interface above for constructing it from the `Int32` / checking if it's non-zero.

Reply:
I was going for the slightly cleaner error messaging we get from making it our own enum. However, if it seems like overkill, I'm happy to remove it